src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -0,0 +1,749 @@
+                10,         // 10^64
+                18,         // 10^128
+                33,         // 10^256
+            };
+
+            private static readonly uint[] Pow10BigNumTable = new uint[]
+            {
+                // 10^8
+                1,          // _length
+                100000000,  // _blocks
+
+                // 10^16
+                2,          // _length
+                0x6FC10000, // _blocks
+                0x002386F2,
+
+                // 10^32
+                4,          // _length
+                0x00000000, // _blocks
+                0x85ACEF81,
+                0x2D6D415B,
+                0x000004EE,
+
+                // 10^64
+                7,          // _length
+                0x00000000, // _blocks
+                0x00000000,
+                0xBF6A1F01,
+                0x6E38ED64,
+                0xDAA797ED,
+                0xE93FF9F4,
+                0x00184F03,
+
+                // 10^128
+                14,         // _length
+                0x00000000, // _blocks
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x2E953E01,
+                0x03DF9909,
+                0x0F1538FD,
+                0x2374E42F,
+                0xD3CFF5EC,
+                0xC404DC08,
+                0xBCCDB0DA,
+                0xA6337F19,
+                0xE91F2603,
+                0x0000024E,
+
+                // 10^256
+                27,         // _length
+                0x00000000, // _blocks
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x982E7C01,
+                0xBED3875B,
+                0xD8D99F72,
+                0x12152F87,
+                0x6BDE50C6,
+                0xCF4A6E70,
+                0xD595D80F,
+                0x26B2716E,
+                0xADC666B0,
+                0x1D153624,
+                0x3C42D35A,
+                0x63FF540E,
+                0xCC5573C0,
+                0x65F9EF17,
+                0x55BC28F2,
+                0x80DCC7F7,
+                0xF46EEDDC,
+                0x5FDCEFCE,
+                0x000553F7,
+
+                // Trailing blocks to ensure MaxBlockCount
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+                0x00000000,
+            };
+
+            private static readonly uint[] MultiplyDeBruijnBitPosition = new uint[]
+            {
+                0, 9, 1, 10, 13, 21, 2, 29, 11, 14, 16, 18, 22, 25, 3, 30,
+                8, 12, 20, 28, 15, 17, 24, 7, 19, 27, 23, 6, 26, 5, 4, 31
+            };
+
+            private int _length;
+            private BlocksBuffer _blocks;
+
+            public BigInteger(uint value)
```

Comment:
Native code can configure the default constructor to efficiently initialize this, rather than requiring us to specialize case `value == 0`